plugins/methodinfo/src/main/java/org/qbicc/plugin/methodinfo/MethodInformation.java

```diff
@@ -0,0 +1,503 @@
+        Path modulePath = emitMethodData(context);
+        if (modulePath != null) {
+            compileMethodDataModule(context, modulePath);
+        }
+    }
+
+    private static class StackMapRecord implements Comparable {
+        private final int objectFileIndex;
+        private final int offset;
+        private final long statepoindId;
+
+        StackMapRecord(final int objectFileIndex, final int offset, final long statepoindId) {
+            this.objectFileIndex = objectFileIndex;
+            this.offset = offset;
+            this.statepoindId = statepoindId;
+        }
+
+        long getStatepoindId() { return statepoindId; }
+        int getOffset() { return offset; }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(objectFileIndex, statepoindId, offset);
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+
+            return equals((StackMapRecord)o);
+        }
+
+        public boolean equals(StackMapRecord record) {
+            return objectFileIndex == record.objectFileIndex
+                && offset == record.offset;
+        }
+
+        @Override
+        public int compareTo(Object o) {
+            StackMapRecord other = (StackMapRecord) o;
+            if (equals(other)) {
+                return 0;
+            }
+            if (objectFileIndex < other.objectFileIndex) {
+                return -1;
+            } else if (objectFileIndex > other.objectFileIndex) {
+                return 1;
+            } else if (offset < other.offset) { // if same objectFileIndex then order by offset
+                return -1;
+            } else {
+                return 1;
+            }
+        }
+    }
+
+    private final class StackMapRecordCollector {
+        CompilationContext context;
+        StackMapRecordCollector(CompilationContext context) {
+            this.context = context;
+        }
+
+        public List<StackMapRecord> collect() {
+            Linker linker = Linker.get(context);
+            List<StackMapRecord> recordList = new ArrayList<>();
+            ObjectFileProvider objFileProvider = context.getAttachment(Driver.OBJ_PROVIDER_TOOL_KEY);
+            Iterator<Path> objFileIterator = linker.getObjectFilePaths().iterator();
+            final Integer[] index = { 0 };
+
+            context.runParallelTask(ctxt -> {
+                Path objFile;
+                for (;;) {
+                    Integer objFileIndex;
+                    synchronized (objFileIterator) {
+                        if (!objFileIterator.hasNext()) {
+                            return;
+                        }
+                        objFile = objFileIterator.next();
+                        objFileIndex = index[0];
+                        index[0] += 1;
+                    }
+                    ObjectFile objectFile = null;
```

Comment:
This should be opened using try-with-resources.

Reply:
Done.